user/tests/wiring/ble/heartrate_central/application.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this program; if not, see <http://www.gnu.org/licenses/>.
+  ******************************************************************************
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "application.h"
+
+BLEScanResult    results[10];
+
+BLEPeerAttribute heartrate;
+BLEPeerDevice    peer;
+
+void heartrateUpdated(uint8_t* data, size_t len) {
+
+}
+
+void setup() {
+}
+
+void loop() {
+    if (BLE.connected()) {
+        int newHr;
+        heartrate->getValue(&newHr);
+    }
+    else {
+        size_t count = BLE.scan(results, 10);
+
+        if (count > 0) {
+            for (uint8_t i = 0; i < count; i++) {
+                bool found = results[i].find(BLE_SIG_AD_TYPE_FLAGS);
+                if (found) {
+                    peer = BLE.connect(results[i].address());
+
+                    if (BLE.connected(peer)) {
+                        peer->attribute("heartrate", &heartrate);
+                        heartrate->onDataReceived(heartrateUpdated);
+                    }
+                }
+
+                break;

```

Comment:
This always breaks when `i` is `0`. The code should be better. In this implementation there is no need for a `for` loop. If you intend to only use the first search result, then the code should make that clear. If, on the other hand, you want to iterate through multiple matching results then you need a `for` loop.

Reply:
The point here is that we need to provide the best practice example in these tests that everyone can copy-paste and follow, so we don't want to show users a buggy pattern that they will copy repeatedly.

Reply:
I see in the SDD you moved the `break` up so it breaks after the first relevant result is found. 👍 Do that here, and probably add a comment like "find first matching result"